mfr/server/static/js/mfr.child.hypothesis.js

```diff
@@ -62,7 +62,23 @@
             window.document.head.appendChild(script);
             window.document.body.classList.add('show-hypothesis');
             hypothesisLoaded = true;
-        });
 
+            var sidePanelOpened = false;
+            var sendAnalyticsIfExpanded = function (expanded) { 
+                if (expanded && !sidePanelOpened) {
+                    sidePanelOpened = expanded;
+                    ga('send', 'event', {
+                        eventCategory: 'Hypothesis',
+                        eventAction: 'Open Hypothesis Panel',
+                        eventLabel: window.DEFAULT_URL.split('/')[5],
```

Comment:
What is this value?  Could you add a comment saying what this is expected to be?

Reply:
`eventLabel` should be the preprint guid.

Reply:
A follow-up question, is it guaranteed that the `GUID` is always at index 5? Can we add some extra check before `ga` to parse `window.DEFAULT_URL.split('/')[5]` and handle unexpected URL format. In addition, having a comment on what the expected URL looks like will be very helpful as well.

Reply:
There is no guarantee that `window.DEFAULT_URL.split('/')[5]` always return the preprint guid. However, if the rendered file is a preprint's primary file, `window.DEFAULT_URL` should be the WB link to the file on OSFStorage (e.g. https://files.us.staging2.osf.io/v1/resources/twq6d/providers/osfstorage/5c9cd1e3c8f5970017ed07d3?direct=&mode=render), and from which we can extract the guid of the preprint.
And since hypothesis is only enabled on a preprint's primary file, it should be fine.
I'll add some checks for unexpected URL and log to sentry when that happens.

Reply:
Sounds good! Thanks for the explanation. This WB API rarely changes. If it ever changed, we would be informed right away.

Reply:
@cslzchen PR updated.

Reply:
I added a TODO comment to your code. Although the `split` you use here is unaware of the HTTP protocol, URL domain and path, I think for now it is good enough. A better way (probably a future improvement) is to fully parse and validate the URL.